modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFolderItemLocalServiceTest.java

```diff
@@ -0,0 +1,355 @@
+		// unlinked
+
+		_objectFolderItemLocalService.deleteObjectFolderItem(
+			_objectDefinition1.getObjectDefinitionId(),
+			_objectFolderB.getObjectFolderId());
+
+		Assert.assertNotNull(
+			_objectFolderItemLocalService.getObjectFolderItem(
+				_objectDefinition1.getObjectDefinitionId(),
+				_objectFolderB.getObjectFolderId()));
+
+		// ObjectFolderItem is deleted when is linked and related object
+		// definition is linked
+
+		ObjectFolderItem objectFolderItem =
+			_objectFolderItemLocalService.addObjectFolderItem(
+				TestPropsValues.getUserId(),
+				_objectDefinition4.getObjectDefinitionId(),
+				_objectFolderB.getObjectFolderId(), 0, 0);
+
+		_objectFolderItemLocalService.deleteObjectFolderItem(objectFolderItem);
+
+		Assert.assertNull(
+			_objectFolderItemLocalService.fetchObjectFolderItem(
+				objectFolderItem.getObjectFolderItemId()));
+	}
+
+	@Test
+	public void testDeleteObjectFolderItemByObjectDefinitionId() {
+		Assert.assertNotNull(
+			_objectFolderItemLocalService.
+				getObjectFolderItemsByObjectDefinitionId(
+					_objectDefinition1.getObjectDefinitionId()));
+
+		_objectFolderItemLocalService.
+			deleteObjectFolderItemByObjectDefinitionId(
+				_objectDefinition1.getObjectDefinitionId());
+
+		Assert.assertEquals(
+			Collections.emptyList(),
+			_objectFolderItemLocalService.
+				getObjectFolderItemsByObjectDefinitionId(
+					_objectDefinition1.getObjectDefinitionId()));
+	}
+
+	@Test
+	public void testDeleteObjectFolderItemByObjectFolderId() {
+		Assert.assertNotNull(
+			_objectFolderItemLocalService.getObjectFolderItemsByObjectFolderId(
+				_objectFolderA.getObjectFolderId()));
+
+		_objectFolderItemLocalService.deleteObjectFolderItemByObjectFolderId(
+			_objectFolderA.getObjectFolderId());
+
+		Assert.assertEquals(
+			Collections.emptyList(),
+			_objectFolderItemLocalService.getObjectFolderItemsByObjectFolderId(
+				_objectFolderA.getObjectFolderId()));
+	}
+
+	@Test
+	public void testUpdateObjectFolderItem() throws PortalException {
+		long objectDefinitionId = RandomTestUtil.randomLong();
+		long objectFolderId = RandomTestUtil.randomLong();
+
+		AssertUtils.assertFailure(
+			NoSuchObjectFolderItemException.class,
+			StringBundler.concat(
+				"No ObjectFolderItem exists with the key {objectDefinitionId=",
+				objectDefinitionId, ", objectFolderId=", objectFolderId, "}"),
+			() -> _objectFolderItemLocalService.updateObjectFolderItem(
+				objectDefinitionId, objectFolderId, 1, 1));
+
+		ObjectFolderItem objectFolderItem =
+			_objectFolderItemLocalService.updateObjectFolderItem(
+				_objectDefinition1.getObjectDefinitionId(),
+				_objectFolderA.getObjectFolderId(), 1, 1);
+
+		Assert.assertEquals(1, objectFolderItem.getPositionX());
+		Assert.assertEquals(1, objectFolderItem.getPositionY());
+	}
+
+	@Test
+	public void testUpdateObjectFolderObjectFolderItem()
+		throws PortalException {
+
+		_objectDefinitionLocalService.updateObjectFolderId(
```

Comment:
I'm thinking if we should inline this method 🤔

Reply:
This method reflect a method existing in LocalService, [see](https://github.com/liferay/liferay-portal/blob/8219becef1c2a9ed47827298da92fc970fb25116/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFolderItemLocalServiceImpl.java#L173-L199)